app/main.py

```diff
@@ -1,23 +1,37 @@
 import init_django_orm  # noqa: F401
 
-from django.db.models import QuerySet
+from django.db.models import QuerySet, Q, F
+from db.models import LoyaltyProgram, LoyaltyProgramParticipant, Customer
 
 
 def all_loyalty_program_names() -> QuerySet:
-    pass
+    return LoyaltyProgram.objects.values_list("name", "bonus_percentage")
 
 
 def not_active_customers() -> QuerySet:
-    pass
+    return LoyaltyProgramParticipant.objects.filter(
+        last_activity__gte="2021-01-01",
+        last_activity__lt="2022-01-01"
```

Comment:
you could use 'last_activity__year' here .

Reply:
In the task we have a condition about the period, and not just about the year